examples/opinion_spread.jl

```diff
@@ -0,0 +1,92 @@
+# # Opinion spread
+
+# This is a simple model of how an opinion spreads through a community. Each individual has a number of opinions as a list of integers. They can change their opinion by changing the numbers in the list.
+
+# Agents can change their opinion at each step. They choose one of their neighbors randomly, and adopt one of the neighbor's opinion. They are more likely to adopt their neighbors opinion if the share more opinions with each other.
+
+using Agents
+using AgentsPlots
+using Random
+
+# ## Building the model
+# ### 1. Model creation
+
+mutable struct Citizen <: AbstractAgent
+	id::Int
+	pos::Tuple{Int, Int}
+	stabilized::Bool
+	opinion::Array{Int, 1}
+	prev_opinion::Array{Int, 1}
+end
+
+function create_model(;dims=(10, 10), nopinions=3, levels_per_opinion=4)
+	space = GridSpace(dims, periodic=true, moore=true)
+	properties = Dict(:nopinions=>nopinions) 
+	model = AgentBasedModel(Citizen, space, scheduler=random_activation, properties=properties)
+	for cell in 1:nv(model)
+		add_agent!(cell, model, false, rand(1:levels_per_opinion, nopinions), rand(1:levels_per_opinion, nopinions))
+	end
```

Comment:
`fill_space!` could be used here

Reply:
I want each agent to have a different initial population, but `fill_space!` uses one set of random numbers for all agents.

Reply:
You can give a function of `position` to `fill_space!`.

Reply:
> You can give a function of position to fill_space!.

Good point. But then the loop is simpler, I guess.